api/models/weather.py

```diff
@@ -1,11 +1,92 @@
 from schematics.models import Model
 from schematics.types import FloatType, GeoPointType, ModelType, StringType
+from schematics.types.serializable import serializable
+from unidecode import unidecode
+from api.utils.metatypes import EnumMeta
 
 class PhysicalQuantity(Model):
     value = FloatType(required=True)
     unit = StringType(required=True)
 
 
+class Temperature(Model):
+    value = FloatType(required=True)
+
+    class TemperatureMetrics(StringType, metaclass=EnumMeta):
+        CELSIUS = 'CELSIUS'
+        FAHRENHEIT = 'FAHRENHEIT'
+        KELVIN = 'KELVIN'
+
+    @serializable(type=FloatType, serialized_name='value')
+    def get_temperature(self, *args, **kwargs):
+        if hasattr(self, 'context'):
+            temperature = self.context.get('temperature', 'kelvin')
+
+            if temperature.upper() == self.TemperatureMetrics.CELSIUS:
+                value = round(self.value - 273.0, 2) # Converts Kelvin temperature to Celsius
+            elif temperature.upper() == self.TemperatureMetrics.FAHRENHEIT:
+                value = round(1.8*(self.value-273.0)+32.0, 2) # Converts Kelvin temperature to Fahrenheit
+            else:
+                value = round(self.value, 2) # Returns Kelvin temperature
+        else:
+            value = round(self.value, 2) # Returns Kelvin temperature
+        return value
+
+    @serializable(type=StringType, serialized_name='unit')
+    def get_unit(self, *args, **kwargs):
+        if hasattr(self, 'context'):
+            unit = self.context.get('temperature', 'kelvin')
+
+            if unit.upper() == self.TemperatureMetrics.CELSIUS:
+                unit = self.TemperatureMetrics.CELSIUS
+            elif unit.upper() == self.TemperatureMetrics.FAHRENHEIT:
+                unit = self.TemperatureMetrics.FAHRENHEIT
+            else:
+                unit = self.TemperatureMetrics.KELVIN
+        else:
+            unit = self.TemperatureMetrics.KELVIN
+        return unit.lower()
+
+
+class Distance(Model):
```

Comment:
Maybe `Distance` and `Temperature` could inherit from an generic abstract "Number with unit" data type, leaving only the metrics, the default one and the transformation functions between each pair (or to and from default) to be defined in each of these.

That would add quite a bit of complexity though and would only be justifiable for production code if there were to exist more classes like these.